crates/server/src/authority/error.rs

```diff
@@ -13,9 +13,9 @@ use thiserror::Error;
 use crate::proto::op::ResponseCode;
 use crate::proto::rr::{rdata::SOA, Record};
 use crate::proto::{ProtoError, ProtoErrorKind};
-#[cfg(feature = "hickory-recursor")]
+#[cfg(feature = "recursor")]
```

Comment:
This is great -- I suggest you change the features:

```toml
recursor = ["hickory-recursor"]
resolver = ["hickory-resolver"]
```

to

```toml
recursor = ["dep:hickory-recursor"]
resolver = ["dep:hickory-resolver"]
```

which should disable the implicit features named after the crates, generating warnings if they are used in `cfg` attributes.